gpbft/gpbft.go

```diff
@@ -1045,40 +1046,53 @@ type convergeState struct {
 	// Chains indexed by key
 	values map[ChainKey]ECChain
 	// Tickets provided by proposers of each chain.
-	tickets map[ChainKey][]Ticket
+	tickets map[ChainKey][]ActorTicket
+}
+
+type ActorTicket struct {
+	Actor  ActorID
+	Ticket Ticket
 }
 
 func newConvergeState() *convergeState {
 	return &convergeState{
 		values:  map[ChainKey]ECChain{},
-		tickets: map[ChainKey][]Ticket{},
+		tickets: map[ChainKey][]ActorTicket{},
 	}
 }
 
 // Receives a new CONVERGE value from a sender.
-func (c *convergeState) Receive(value ECChain, ticket Ticket) error {
+func (c *convergeState) Receive(sender ActorID, value ECChain, ticket Ticket) error {
 	if value.IsZero() {
 		return fmt.Errorf("bottom cannot be justified for CONVERGE")
 	}
 	key := value.Key()
 	c.values[key] = value
-	c.tickets[key] = append(c.tickets[key], ticket)
+
+	c.tickets[key] = append(c.tickets[key], ActorTicket{Actor: sender, Ticket: ticket})
```

Comment:
Please leave a comment referencing https://github.com/filecoin-project/go-f3/issues/12 that we need to drop duplicates rather than accumulate an unbounded list of them.

Reply:
Done, thanks.